crates/telemetry/src/config.rs

```diff
@@ -0,0 +1,68 @@
+use envconfig::Envconfig;
+
+/// Provides configuration for the telemetry system.
+///
+/// Largely consists of OpenTelemetry variables defined
+/// [here](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration)
+/// but also has some Spin specific configuration.
+#[derive(Debug)]
+pub struct Config {
+    /// Disables OpenTelemetry for all signals (traces and metrics).
+    pub otel_sdk_disabled: bool,
+
+    /// Determines the verbosity of the OpenTelemetry tracing layer.
+    ///
+    /// This is a Spin specific value. This is what allows us to have different levels of verbosity
+    /// in the fmt layer and the otel layer. The fmt layer respects `RUST_LOG` and the otel layer
+    /// respects `SPIN_OTEL_TRACING_LEVEL`.
+    pub otel_tracing_level: String,
+
+    /// Spin version.
+    pub version: String,
+}
+
+impl Config {
+    /// Initializes a new [Config] from the environment.
+    pub fn from_env(version: String) -> anyhow::Result<Self> {
+        let inner = EnvDerivableConfig::init_from_env()?;
+        Ok(Self {
+            otel_sdk_disabled: inner.otel_sdk_disabled,
+            otel_tracing_level: inner.otel_tracing_level,
+            version,
+        })
+    }
+
+    /// Returns a boolean indicating if the OTEL layer should be enabled.
+    ///
+    /// It is considered enabled if any of the following environment variables are set:
```

Comment:
If I set one of these but also set `otel_sdk_disabled`, does the latter take precedence?

(ETA: reading elsewhere in the code, it appears yes.)